NeoMathEngine/src/MemoryEngine.h

```diff
@@ -0,0 +1,66 @@
+/* Copyright © 2024 ABBYY
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <NeoMathEngine/NeoMathEngineDefs.h>
+#include <RawMemoryManager.h>
+#include <memory>
+#include <mutex>
+
+namespace NeoML {
+
+class CDeviceStackAllocator;
+class CHostStackAllocator;
+class CMemoryPool;
+
+// Memory management engine base class
+class IMemoryEngine : public IMathEngine {
```

Comment:
Not an interface - it is base class.
I propose name `CMemoryEngineMixin`

Reply:
fixed